pkg/apiclient/grpcproxy.go

```diff
@@ -113,7 +113,18 @@ func (c *client) startGRPCProxy() (*grpc.Server, net.Listener, error) {
 			if err != nil {
 				return err
 			}
+
 			md, _ := metadata.FromIncomingContext(stream.Context())
+
+			if len(c.AdditionalHeaders) > 0 {
+				for _, kv := range c.AdditionalHeaders {
```

Comment:
as above - either you should have a test, or you should not bother with this check